### ISSUE TITLE
Python backend 0-size GPU tensors causing "failed to get cuda pointer device attribute: invalid argument"

### ISSUE THREAD

**Issue by darintay** [triton-inference-server/server#4865]
**Description**
Returning a gpu-backed tensor where one of the dimensions is size 0 causes the inference to fail with the error "message: failed to get cuda pointer device attribute: invalid argument".

This only occurs if the tensor is on the GPU, and one of its dimensions are 0.

I've put together a minimal example below.

**Triton Information**

What version of Triton are you using?
22.08

Are you using the Triton container or did you build it yourself?
Container, from NGC.

**To Reproduce**
Steps to reproduce the behavior.

1. Setup model repo

config.pbtxt
```
name: "test_model"
backend: "python"

input [
  {
    name: "X"
    data_type: TYPE_FP32
    dims: [ 4 ]
  }
]
output [
  {
    name: "OUTPUT"
    data_type: TYPE_FP32
    dims: [ -1 ]
  }
]
```

model.py
```python
import torch
import triton_python_backend_utils as pb_utils
from torch.utils.dlpack import to_dlpack, from_dlpack


class TritonPythonModel:
    def initialize(self, args):
        pass

    def execute(self, requests):
        responses = []

        for request in requests:
            in_0 = pb_utils.get_input_tensor_by_name(request, "X")

            SHAPE = (0, )

            # Adding this line avoids a crash.
            # SHAPE = (1, )

            torch_test = torch.ones(SHAPE, dtype=torch.float32)
            device = torch.device("cuda:0")
            torch_test = torch_test.to(device)

            # Adding this line avoids a crash.
            # torch_test = torch_test.to("cpu")

            out_tensor_1 = pb_utils.Tensor.from_dlpack("OUTPUT", to_dlpack(torch_test))

            inference_response = pb_utils.InferenceResponse(
                output_tensors=[out_tensor_1]
            )
            responses.append(inference_response)

        return responses
```

2. Run in the NGC container:
```bash
$ pip install torch
$ tritonserver --model-repository .
$ perf_analyzer -m test_model
...

0910 00:55:16.004736 357 pb_stub.cc:777] Failed to process the request(s) for model 'test_model', message: failed to get cuda pointer device attribute: invalid argument
```

3. Note that uncommenting either of the commented out lines (non-0-dim or cpu placement) causes the model to start working again.

**Expected behavior**
Should correctly return the tensor output.

Instead, it errors (unless you place the tensor on CPU, or ensure all dimensions are non-0).

**Comment by kthui** [comment]
Hi @darintay, I have filed a ticket for us to look further into this issue.
